javascript-challenges.test.js

```diff
@@ -14,30 +14,98 @@
+  }
+
 
 // --------------------2) Create a function that takes in an array of 5 numbers and determines whether or not the array is a "full house". A full house is exactly one pair and one three of a kind.
 
 // a) Create a test with expect statements using the variable provided.
 
-const hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
-const hand4 = [7, 2, 7, 2, 7]
-// Expected output: true
 
-// b) Create the function that makes the test pass.
 
-// Pseudo code:
+// Create a test function with the name in double quotes, ending an anonymous function
+describe("fullHouse", () => {
+    // Create another function with an it statement, followed up with the original prompt and another anonymous function
+    it("takes in an array of 5 numbers and determines whether or not the array is a full house.", () => {
+        const hand1 = [5, 5, 5, 3, 3]
+        // Expected output: true
+        const hand2 = [5, 5, 3, 3, 4]
+        // Expected output: false
+        const hand3 = [5, 5, 5, 5, 4]
+        // Expected output: false
+        const hand4 = [7, 2, 7, 2, 7]
+        // Expected output: true
+        // expect function name (variable name) to equal (output)   
+        expect(fullHouse(hand1)).toEqual(true)    
+        // expect function name (variable name) to equal (output)   
+        expect(fullHouse(hand2)).toEqual(false)    
+        // expect function name (variable name) to equal (output)   
+        expect(fullHouse(hand3)).toEqual(false)    
+        // expect function name (variable name) to equal (output)   
+        expect(fullHouse(hand4)).toEqual(true)    
+  })
+})
+
+// // ReferenceError: fullHouse is not defined
+
+// // b) Create the function that makes the test pass.
+
+// Pseudo code: 
+// Function called fullHouse with one parameter of hand
+const fullHouse = (hand) => {
+    // Function called frequency to hold an empty array
+    const frequency = []
+    // loop for each number in the array
+    hand.forEach(number => {
+        // check for frequency of each number in the array and store it in the frequency variable, incrementing by count of 1
+        // logical OR || it takes a zero 0
+        frequency[number] = (frequency[number] || 0) + 1 // Personal note: number parameter is in square brackets b/c its a number
+    })
+    // since arrays are objects, I can use the Object method.
+    // values is equal to the values of the Object, frequency.
+    const values = Object.values(frequency)
+    // return the included pair (2) values AND the included 3 of a kind (3) values
+    // if both conditions are true it will return true, otherwise it will return false.
+    return values.includes(2) && values.includes(3)
+}
+
```

Comment:
Jackson, based on your request...I think either you're overthinking it, not recognizing that you know more than you do, or maybe not breaking down the problems far enough.  Probably a combination of all 3 😁.  There's no need for regex or anything like that in the first one. Although not as sexy, you can use a conditional statement and reassign the letters when appropriate. The second one, think about what the first thing you do when you get your hand when playing cards.  Then think about the win conditions.  Again, conditional will take care of it.  More code isn't always a bad thing.  Don't get hung up on the oneliners and such.  I've seen you make so much progress.  You can do this!